### PR TITLE
Oleg Shtynda

### PR DESCRIPTION
**Pull request by OlegShtynda**
deletable buttons on targets, front and back end for banks

### CODE REVIEW

File: FundTrack.BLL/Concrete/TargetService.cs [hunk 1]
```diff
@@ -43,9 +44,106 @@ public IEnumerable<TargetViewModel> GetTargetsByOrganizationId(int id)
...
+        {
+            CheckInTargets(targets);
+            CheckInOrgAccounts(targets);
+            CheckInDonations(targets);
+            CheckInFinOps(targets);
+        }
+
+        private void CheckInTargets(List<TargetViewModel> targets)
+        {
+            foreach (var target in targets)
```
Comment (Barabashka88) [inline FundTrack.BLL/Concrete/TargetService.cs]: can be improved by reducing array of targets only for subtargets

File: FundTrack.BLL/Concrete/TargetService.cs [hunk 2]
```diff
@@ -43,9 +44,106 @@ public IEnumerable<TargetViewModel> GetTargetsByOrganizationId(int id)
...
+            {
+                throw new BusinessLogicException(ErrorMessages.CantFindDataById, e); 
+            }
+        }
+
+        private void CheckInFinOps(List<TargetViewModel> targets)
+        {
+            try
+            {
+                var finOps = _unitOfWork.FinOpRepository.Read().ToList();
```
Comment (BogdanLyseiko) [inline FundTrack.BLL/Concrete/TargetService.cs]: add where for get finops for current organization

File: FundTrack.DAL/Concrete/BankRepositoty.cs [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+            return _context.Banks
+                .FirstOrDefault(b => b.Id == bankId);
+        }
+
+        public Bank Create(Bank item)
+        {
+            return _context.Banks.Add(item).Entity;
+        }
+
+        public Bank Update(Bank bank)
```
Comment (BogdanLyseiko) [inline FundTrack.DAL/Concrete/BankRepositoty.cs]: in this method you don't  write  data to database.
use context.Banks.Update method

File: FundTrack.WebUI/ClientApp/app/components/finance/createorgaccount.component.ts [hunk 4]
```diff
@@ -34,6 +35,16 @@ export class CreateOrgAccountComponent {
     private smallAccountForm: FormGroup;
     // Property that keeps all targets of current organization
     private targets: TargetViewModel[] = new Array<TargetViewModel>();
+    private defaultTarget: TargetViewModel = {
+        targetId: undefined,
+        name: "Не вказано",
```
Comment (BogdanLyseiko) [inline FundTrack.WebUI/ClientApp/app/components/finance/createorgaccount.component.ts]: move to separate file with constants

File: FundTrack.WebUI/ClientApp/app/components/finance/createorgaccount.component.ts [hunk 5]
```diff
@@ -89,42 +101,39 @@ export class CreateOrgAccountComponent {
...
-    private numberMessage = "Поле повинно містити тільки цифри";
-    private LengthMessage = "Недопустима кількість символів";
-    private accountNameWrongLength = "Максимально допустима довжина імені рахунку складає 100 символів";
-    private mfolength = "МФО повинно містити 6 цифр";
-    private edrpoulength = "ЄДРПОУ повинно містити 8 цифр";
-    private banknamelength = "Ім'я банку не може перевищувати 50 символів";
-    private descriptionlength = "Опис рахунку не може перевищувати 200 символів";
-    private currencyselection = "Необхідно вибрати валюту";
-    private wrongBalanceMessage = "Баланс повинен бути у форматі 1 або 1.23";
+    private requiredMessage = "*Поле є обов'язковим для заповнення";
```
Comment (BogdanLyseiko) [inline FundTrack.WebUI/ClientApp/app/components/finance/createorgaccount.component.ts]: move to separate file with constants

Comment (BogdanLyseiko) [inline FundTrack.WebUI/ClientApp/app/components/finance/createorgaccount.component.ts]: move to separate file with constants

File: FundTrack.WebUI/ClientApp/app/services/concrete/finance/orgaccount.service.ts [hunk 6]
```diff
@@ -170,4 +172,14 @@ export class OrgAccountService {
             .map((res: Response) => <DonateCredentialsViewModel>res.json())
             .catch((error: Response) => this.handleError(error));
     }
+
+    public getAllBanks(): Observable<BankViewModel[]> {
+        return this._http.get(this._getBanksUrl)
```
Comment (BogdanLyseiko) [inline FundTrack.WebUI/ClientApp/app/services/concrete/finance/orgaccount.service.ts]: move url to GlobalUrlService

**Pull request merged**